.github/workflows/docker-publish.yml

```diff
@@ -0,0 +1,33 @@
+name: Publish Docker image
+on:
+  release:
+    types: [published]
+  push:
+    branches:
+      - master
+      - dev
+  pull_request:
+    branches: [dev]
+
+jobs:
+  push_to_registries:
+    name: Push Docker Image to DockerHub and Github Packages
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check out the repo
+        uses: actions/checkout@v2
+      - name: Push to Docker Hub
+        uses: docker/build-push-action@v1
+        with:
+          username: ${{ secrets.DOCKER_USERNAME }}
```

Comment:
Where will these secrets reside?

Reply:
Secrets used by GitHub Action workflows are stored on GitHub. They can be accessed/set by admin under the settings page for aa given repo. In this particular case the `DOCKER_USERNAME` and `DOCKER_PASSWORD` are my personal userrname/password because my acm account is linked as admin for our docker hub. You can try setting secrets for any public/private repo in your personal GitHub to see what its like. These secrets are essentially a way to ensure that our CI/CD workflows can run without exposing important secrets. 

While we're on the topic of secrets there are a couple more additions that need to be made as part of a separate PR. The current containerization process doesn't pass the required variables to the image which is something that needs to be fixed. The current docker image that gets built has all the env variables set to undefined. 

Once we setup a clean pipeline for passing secrets to Google Cloud Build & our CI/CD processes running within GCP we will use [Secret Manager](https://cloud.google.com/secret-manager).